migrate to gen.coroutine

vilisov opened the pull request.

Commit:
added 2to3 for setup.py

Commit:
fix for python3

Commit:
fix EmailField bug with trying to validate default=None

Commit:
find and aind_on changes (@gen.engine -> @gen.coroutine)

Commit:
all method for Manager

Commit:
small refactoring collection.py

Commit:
count() over gen.coroutine

Commit:
incr version

tahajahangir commented on mongotor/orm/collection.py:
```diff
@@ -32,6 +33,9 @@
     def __new__(cls, name, bases, attrs):
+        if not attrs.get('__collection__'):
+            attrs['__collection__'] = re.sub(
+                '([A-Z]+)', r'_\1', name).lower()[1:]
```
This is correct but normal upper-camel-case class names, but omits first character if class name starts with lower-case letter.

tahajahangir commented:
I reviewed the changes, most of them seems OK, but this notes should be considered:
- using bytes literal (`b'data'`) breaks python 2.5 compatibility (no-one is using python2.5 of-course), documentation should be updated to mention support of  python2.6+.
- The one line function `Collection.get_client` is removed (and inlined), but it's a backward-incompatible change and also is not a necessary. I think this change should be reverted.
- Name of package (in `setup.py`) should change back to `mongotor` (from `mongotor-skd`)
- New method `Collection.all` added, documentation needed.
- `Collection.__collection__` attribute is now auto-generated from class name (if not explicitly defined) by converting upper-camel-case style to underline-separated style. This is a good change, but docs is not updated.

Commit:
fix bug with __collection__ autofill

Commit:
add documentation for __collection__ auto-generation

Commit:
add documentation for manager's all method

Commit:
add python 2.6+ requirement, change package name back to mongotor

ghost commented:
I do not understand about `get_client` method. swight did not remove get_client, he add it.

tahajahangir commented:
Yes, I was wrong about `get_client` method! everything is ok.

Commit:
fix bug with endless recursion
when call hasattr(self, '_initialized') with not initialized database

Commit:
fix collection tests

Commit:
distinct over gen.coroutine

ghost commented:
@tahajahangir could you please fix two tests in `test_signal.py`? I have no idea why they fails.

Commit:
fix bug (mistype) with max_value validation in NumberField

Commit:
fix bug with NumberField
when default is not specified but min or max value is specified
you got a validation error

Commit:
add remove method to collection manager

Commit:
complete manager over gen.coroutine

tahajahangir commented:
I rebased the changes to add travis build support. The new branch is located in `tahajahangir/coroutine-py3`.

I fixed the two failing tests in orm.signals (they incorrectly used two nested stop/wait calls). The good news is tests are passing in python 2.7, and the bad news is they passed only on python 2.7 (failed in 2.6, 3.2, 3.3, 3.4 and pypy). Build results is accessible in https://travis-ci.org/tahajahangir/mongotor/builds/26819709 . 

@marcelnicolay : What's the reason to fix versions in `requirements-dev.txt`? It seems builds on python 3.x and pypy is failed because of outdated required packages.

@geerk : Can you fix syntax errors on python 2.6? (see [build status](https://travis-ci.org/tahajahangir/mongotor/jobs/26819710))

ghost commented:
@tahajahangir why we should support 2.6? I think those who use Tornado and MongoDB will not use 2.6.

tahajahangir commented:
I added [some commits](https://github.com/tahajahangir/mongotor/compare/coroutine-py3) (mainly added use of six and removed use of sure) and tests now run on python 2.7/3.2/3.3/3.4 [(build status)](https://travis-ci.org/tahajahangir/mongotor/builds/26885747)

Only one test is failing on python 2.6. On pyy, there are several failing tests, and test process stucks at `test_load_two_in_pool_connections`.

I also pushed a merge of current open pull requests (#42 #42 #44 #37) to [tahajahangir/master](https://github.com/tahajahangir/mongotor/tree/master)
